### ISSUE TITLE
Create CineSimile Homepage

### ISSUE THREAD

**Issue by Niweera** [Niweera/cinesimile#4]
## CineSimile Homepage
The following is how the CineSimile homepage should be looked like. Keep in mind this is just a suggestion. 

![image](https://s3.amazonaws.com/assets.mockflow.com/app/wireframepro/company/Cbc036d63685af7554822b900549e3552/projects/M04faed3d3caa5e41d9fdefecd14727d51601181423157/pages/1ae2ad9ee80b40d59b90297b3e61f3ab/image/1ae2ad9ee80b40d59b90297b3e61f3ab.png)

Since we use Carbon Design System and React,  we can easily create a single component and integrate them later. The following would be the suggested list of components. If you would like to work on a component and create a pull request, please mention that in the comments. Also, if you want to suggest a new component, please mention that in the comment too.

- [ ] Navbar 
- [ ] Introduction Component
- [ ] Search box 
- [ ] Search results title 
- [ ] Results movie card
- [ ] Footer

## Important

When sending a pull request, please remember to add a screenshot of the development of your selected component. In that case, it is easy for others to get inspired too 😁.

**Comment by VinuriBandara** [comment]
I would like to contribute @Niweera 😊

**Comment by Niweera** [comment]
@VinuriBandara you can select a component that you want to develop and mention it here. I will assign you to that component development.

**Comment by VinuriBandara** [comment]
I would like to work on the results movie cards @Niweera

**Comment by shravani05** [comment]
I would like to work on the Introduction Component @Niweera

**Comment by Niweera** [comment]
Thanks, @VinuriBandara, we'll assign you to results movie component development. 
Thanks, @shravani05, we'll assign you to the Introduction component development.
If you have any questions regarding the development, do not hesitate to ask. 😎
Thanks for contributing to CineSimile. Happy Hacktoberfest! 😁

**Comment by utkarsh4321** [comment]
Hii @Niweera i wanna to work on footer part.

**Comment by Niweera** [comment]
@utkarsh4321 I have assigned you to the task of developing the Footer.

**Comment by MukulTan10** [comment]
hey i would like to work on navbar

**Comment by Niweera** [comment]
@MukulTan10, sure I'll assign you to the task.

**Comment by shravani05** [comment]
I have made the PR with the necessary requirements for the Introduction Component of the Home page.

**Comment by jpalacio0612** [comment]
Hello. 

My name is Jonathan. I am a FullStack Mern Developer,I would like to contribute, I was doing some tests with the part of the search bar, This is what I achieved:

![ciensimile](https://user-images.githubusercontent.com/65427911/94752856-0457da00-0352-11eb-838b-9c0554db46dd.gif)

I created a search box component with a functional approach using hooks like useState and useDebouncedCallback to handle bouncing when written to input. And to simulate the calls to the API I did a mockdata and a simple logic to filter.

Stay tuned if it's okay to make a pull request

```javascript
const SearchBox = () => {
  const [query, setQuery] = useState([]);

  const debounced = useDebouncedCallback((value, setQuery) => {
    let array = [];
    if (value) {
      for (let movie of mockMovies) {
        if (movie.title.toLowerCase().startsWith(value)) {
          array.push(movie);
        }
      }
    }
    setQuery(array);
  }, 500);

  return (
    <div className="searchBox">
      <Search
        onChange={(e) => debounced.callback(e.target.value, setQuery)}
        id="search-1"
        placeHolderText="Search a movie"
      />
      {query.map((movie) => (
        <div key={movie.id}>
          <h1>{movie.title}</h1>
        </div>
      ))}
    </div>
  );
};
```

**Comment by Niweera** [comment]
@jpalacio0612 very nice. If you want I can assign you to the issue. Please let me know.

**Comment by jpalacio0612** [comment]
@Niweera Yes, please assign me the task and let me know if I should add something else to it before doing the pull request.

**Comment by Niweera** [comment]
@jpalacio0612 you can send the pull request. Once it merges I think @utkarsh4321 can take a look and do any improvements as they see fit.

**Comment by kots14** [comment]
Is there any API for filling the auto complete? I see that currently there's dummy data. If there's a scope to contribute for the auto complete feature for the movie search, be it front end or back end, I'd like to work on it.

**Comment by Niweera** [comment]
> Is there any API for filling the auto complete? I see that currently there's dummy data. If there's a scope to contribute for the auto complete feature for the movie search, be it front end or back end, I'd like to work on it.

The back-end API is currently under development and we haven't started it. Currently, we are focused on front-end development right now.

**Comment by MilePaor** [comment]
@Niweera i would like to contribute for component called "Search results title"

**Comment by Niweera** [comment]
@MilePaor sure you can. I'll assign you to it.

**Comment by senshiii** [comment]
Hi @Niweera . I just forked and cloned the repo and came to see two links in the Navbar [ News, Contact Us ] . I want to contribute. I can work on News using an API from RapiApi or  I can also do the contact form. Whichever you assign.

**Comment by Niweera** [comment]
> Hi @Niweera . I just forked and cloned the repo and came to see two links in the Navbar [ News, Contact Us ] . I want to contribute. I can work on News using an API from RapiApi or I can also do the contact form. Whichever you assign.

You can choose anything you like, if you prefer you can do the API implement with RapidAPI. I'll assign you.

**Comment by senshiii** [comment]
Yeah sure. I'll do it then.
